lib/index.js

```
@@ -162,20 +166,42 @@ class MqttProtocol {
       const connectionId = this.connections.get(client);
 
       if (connectionId === undefined) {
+        this.context.debug('[mqtt] no connection id for client id %s', client.id);
         return;
       }
 
+      let payload;
+      try {
+        payload = JSON.parse(packet.payload.toString());
+      }
+      catch (e) {
+        const errReq = new this.context.Request({}, {
+          connectionId,
+          protocol: this.protocol,
+          error: new this.context.errors.BadRequestError(e.message)
+        });
+        return this._respond(client, errReq.response.toJSON());
+      }
+
```

Boyscouting: JSON parse errors were not handled.